src/app/WriteClient.h

```diff
@@ -214,11 +214,11 @@ class WriteClient : public Messaging::ExchangeDelegate
      *  handle calling Shutdown on itself once it decides it's done with waiting
      *  for a response (i.e. times out or gets a response). Client can specify
      *  the maximum time to wait for response (in milliseconds) via timeout parameter.
-     *  Default timeout value will be used otherwise.
+     *  If the timeout is missing or is set to System::Clock::kZero, a value based on the MRP timeouts of the session will be used.
      *  If SendWriteRequest is never called, or the call fails, the API
      *  consumer is responsible for calling Shutdown on the WriteClient.
      */
-    CHIP_ERROR SendWriteRequest(const SessionHandle & session, System::Clock::Timeout timeout = kImMessageTimeout);
+    CHIP_ERROR SendWriteRequest(const SessionHandle & session, System::Clock::Timeout timeout = System::Clock::kZero);
```

Comment:
As a side-effect of this, we could pivot `mResponseTimeout` on the exchange to `mExecutionTimeout`, and assign a default value to that that should work in most scenarios (say, 1s). At the time the timer is armed, it would always tack-on the ACK timeout.

This way, applications/protocols don't actually then have to call `SetResponseTimeout` unless they have a long-running operation on the target (e.g CASE processing) that would need the over-ride. This I think, is a nice simplification as well.